### PR TITLE
Pass a param to postUpdateApplication listener to check whether user claims are updated.

### PR DESCRIPTION
**Pull request by ravinduramesh**
### Proposed changes in this pull request
Pass a param to postUpdateApplication listener to check whether user claims are updated before clearing cache in postUpdateApplication function.

### When should this PR be merged
Merge after merging the following PR and bumping its new version in this repo.
- https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1960

### Related Issues
- https://github.com/wso2/product-is/issues/15127

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 1]
```diff
@@ -2366,20 +2377,27 @@ public void updateApplicationByResourceId(String resourceId, ServiceProvider upd
                 throw buildClientException(APPLICATION_NOT_FOUND, msg);
             }
 
-            String updatedAppName = updatedApp.getApplicationName();
+            ServiceProvider storedSP = getServiceProvider(updatedSP.getApplicationID());
+            String updatedClaimConfigString = new Gson().toJson(updatedSP.getClaimConfig());
+            String storedClaimConfigString = new Gson().toJson(storedSP.getClaimConfig());
+            if (updatedClaimConfigString.equals(storedClaimConfigString)) {
+                isClaimConfigChanged = false;
+            }
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2385]: Can't we move this logic to the specific listener that requires to detect whether the claim configuration has changed?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AbstractApplicationMgtListener.java [hunk 2]
```diff
@@ -50,7 +50,8 @@ public boolean doPreUpdateApplication(ServiceProvider serviceProvider, String te
         return true;
     }
 
-    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+    public boolean doPostUpdateApplication(ServiceProvider serviceProvider, String tenantDomain,
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AbstractApplicationMgtListener.java:53]: -1 for changing this just to indicate the claimConfig has changed.

**Review by mefarazath — CHANGES_REQUESTED** [review]
Can we avoid interface signature changes?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 1 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java:2385]: Addressed these comments in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1963
And closing this PR.

**Pull request closed without merging**